Guide/file-storage.markdown

```diff
@@ -575,3 +575,273 @@ config = do
+
+We start by adding the `cryptonite` package to the `default.nix` file, and re-start `./start` to get the package.
+
+```nix
+# default.nix
+haskellEnv = import "${ihp}/NixSupport/default.nix" {
+    ihp = ihp;
+    haskellDeps = p: with p; [
+        cabal-install
+        base
+        # ...
+        cryptonite # <--- ADD THIS
+    ];
+```
+
+Then we need to generate a private and public key pair. We can do it in our Config file, and save the keys for later use.
+
+```haskell
+-- Config/Config.hs
+import Crypto.PubKey.RSA as RSA
+
+data RsaPublicAndPrivateKeys = RsaPublicAndPrivateKeys { publicKey :: RSA.PublicKey, privateKey :: RSA.PrivateKey }
+
+config :: ConfigBuilder
+config = do
+    -- See https://ihp.digitallyinduced.com/Guide/config.html
+
+    -- ...
+
+    -- Private and public keys to sign and verify image style URLs.
+    (publicKey, privateKey) <- liftIO $ liftIO $ RSA.generate 300 65537
+    option $ RsaPublicAndPrivateKeys publicKey privateKey
+
+```
+
+Add a few helper functions, a few for the Controllers.
+
+```haskell
+-- Application/Helper/Controller.hs
+
+module Application.Helper.Controller where
+
+-- ...
+import Config
+import Crypto.PubKey.RSA as RSA
+import Data.ByteString.Base64 as Base64
+import Crypto.PubKey.RSA.PKCS15 as RSA.PKCS15
+import Crypto.Hash.Algorithms as Hash.Algorithms
+
+
+-- | The RSA public key, can be used to verify image style URLs that were signed.
+rsaPublicKey :: (?context :: ControllerContext) => RSA.PublicKey
+rsaPublicKey = (getAppConfig @Config.RsaPublicAndPrivateKeys).publicKey
+
+-- | The RSA private key, can be used to sign image style URLs.
+rsaPrivateKey :: (?context :: ControllerContext) => RSA.PrivateKey
+rsaPrivateKey = (getAppConfig @Config.RsaPublicAndPrivateKeys).privateKey
+
+rsaSignatureMatches :: (?context :: ControllerContext) =>  Text -> Text -> Bool
+rsaSignatureMatches original signature = case Base64.decode $ cs signature of
+    Left msg -> False
+    Right decodedSignature -> RSA.PKCS15.verify (Just Hash.Algorithms.SHA256) rsaPublicKey (cs original) decodedSignature
+```
+
+And one for the View helper.
+
+```haskell
+-- Application/Helper/View.hs
+module Application.Helper.View where
+
+import IHP.ViewPrelude
+import Crypto.PubKey.RSA.PKCS15 as RSA
+import Crypto.Hash.Algorithms as Hash.Algorithms
+import Data.ByteString.Base64 as Base64
+import Application.Helper.Controller
```

Comment:
The compiler is ok with it - just confirming it's conceptually ok.

Reply:
yes, that should work 👍